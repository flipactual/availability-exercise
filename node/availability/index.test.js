console.error = jest.fn();

const nock = require('nock');

const mock = require('./index.mock');

const availability = require('./');

const BASE = 'https://www.thinkful.com';
const PATH = '/api/advisors/availability';

describe('availability', () => {
  describe('outage', () => {
    nock(BASE)
      .get(PATH)
      .twice()
      .reply(410, undefined);

    it('returns an error', async () => {
      expect(await availability()).toMatchInlineSnapshot(`
                        Object {
                          "error": "Failed to retrieve availability",
                        }
                  `);
    });
    it('logs an error', async () => {
      await availability();
      expect(console.error).toHaveBeenLastCalledWith(
        'Failed to retrieve availability',
        ''
      );
    });
  });
  it('returns available times', async () => {
    nock(BASE)
      .get(PATH)
      .once()
      .reply(200, mock);
    expect(await availability()).toMatchSnapshot();
  });
  it('returns an empty array when there are no available slots', async () => {
    nock(BASE)
      .get(PATH)
      .once()
      .reply(200, {});
    expect(await availability()).toMatchInlineSnapshot(`
      Object {
        "data": Array [],
      }
    `);
  });
});
