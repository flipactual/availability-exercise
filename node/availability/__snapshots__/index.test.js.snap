// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`availability returns available times 1`] = `
Object {
  "data": Array [
    Object {
      "date": "2019-04-30",
      "openings": Array [
        Object {
          "advisor": "319369",
          "slots": Array [
            "2019-04-30T19:00:00-04:00",
          ],
        },
        Object {
          "advisor": "335698",
          "slots": Array [
            "2019-04-30T17:30:00-04:00",
            "2019-04-30T16:15:00-04:00",
            "2019-04-30T11:15:00-04:00",
            "2019-04-30T15:30:00-04:00",
            "2019-04-30T14:30:00-04:00",
          ],
        },
        Object {
          "advisor": "372955",
          "slots": Array [
            "2019-04-30T10:15:00-04:00",
          ],
        },
        Object {
          "advisor": "397688",
          "slots": Array [
            "2019-04-30T15:15:00-04:00",
          ],
        },
        Object {
          "advisor": "399956",
          "slots": Array [
            "2019-04-30T10:30:00-04:00",
            "2019-04-30T02:00:00-04:00",
            "2019-04-30T08:00:00-04:00",
            "2019-04-30T01:00:00-04:00",
          ],
        },
        Object {
          "advisor": "399958",
          "slots": Array [
            "2019-04-30T19:15:00-04:00",
          ],
        },
        Object {
          "advisor": "399959",
          "slots": Array [
            "2019-04-30T12:15:00-04:00",
            "2019-04-30T16:00:00-04:00",
            "2019-04-30T14:15:00-04:00",
            "2019-04-30T13:00:00-04:00",
            "2019-04-30T15:00:00-04:00",
            "2019-04-30T09:00:00-04:00",
          ],
        },
        Object {
          "advisor": "417239",
          "slots": Array [
            "2019-04-30T15:45:00-04:00",
            "2019-04-30T18:15:00-04:00",
            "2019-04-30T17:45:00-04:00",
          ],
        },
        Object {
          "advisor": "419054",
          "slots": Array [
            "2019-04-30T04:00:00-04:00",
            "2019-04-30T03:00:00-04:00",
            "2019-04-30T18:00:00-04:00",
            "2019-04-30T06:00:00-04:00",
            "2019-04-30T16:30:00-04:00",
            "2019-04-30T07:00:00-04:00",
            "2019-04-30T17:00:00-04:00",
            "2019-04-30T05:00:00-04:00",
          ],
        },
        Object {
          "advisor": "424108",
          "slots": Array [
            "2019-04-30T11:00:00-04:00",
            "2019-04-30T10:00:00-04:00",
            "2019-04-30T12:00:00-04:00",
            "2019-04-30T14:00:00-04:00",
          ],
        },
        Object {
          "advisor": "424185",
          "slots": Array [
            "2019-04-30T00:00:00-04:00",
          ],
        },
      ],
    },
    Object {
      "date": "2019-04-28",
      "openings": Array [
        Object {
          "advisor": "335698",
          "slots": Array [
            "2019-04-28T15:00:00-04:00",
            "2019-04-28T16:00:00-04:00",
            "2019-04-28T13:15:00-04:00",
            "2019-04-28T10:00:00-04:00",
            "2019-04-28T12:00:00-04:00",
            "2019-04-28T11:00:00-04:00",
            "2019-04-28T17:00:00-04:00",
            "2019-04-28T14:00:00-04:00",
          ],
        },
        Object {
          "advisor": "399956",
          "slots": Array [
            "2019-04-28T01:00:00-04:00",
          ],
        },
        Object {
          "advisor": "424185",
          "slots": Array [
            "2019-04-28T02:00:00-04:00",
            "2019-04-28T13:00:00-04:00",
            "2019-04-28T09:00:00-04:00",
            "2019-04-28T04:00:00-04:00",
            "2019-04-28T20:00:00-04:00",
            "2019-04-28T23:00:00-04:00",
            "2019-04-28T07:00:00-04:00",
            "2019-04-28T18:00:00-04:00",
            "2019-04-28T00:00:00-04:00",
            "2019-04-28T19:00:00-04:00",
            "2019-04-28T08:00:00-04:00",
            "2019-04-28T06:00:00-04:00",
            "2019-04-28T22:00:00-04:00",
            "2019-04-28T05:00:00-04:00",
            "2019-04-28T03:00:00-04:00",
            "2019-04-28T21:00:00-04:00",
          ],
        },
      ],
    },
    Object {
      "date": "2019-04-29",
      "openings": Array [
        Object {
          "advisor": "319369",
          "slots": Array [
            "2019-04-29T19:00:00-04:00",
          ],
        },
        Object {
          "advisor": "335698",
          "slots": Array [
            "2019-04-29T12:15:00-04:00",
            "2019-04-29T14:00:00-04:00",
            "2019-04-29T17:15:00-04:00",
            "2019-04-29T15:15:00-04:00",
          ],
        },
        Object {
          "advisor": "372955",
          "slots": Array [
            "2019-04-29T10:00:00-04:00",
            "2019-04-29T13:00:00-04:00",
            "2019-04-29T17:30:00-04:00",
          ],
        },
        Object {
          "advisor": "399956",
          "slots": Array [
            "2019-04-29T01:00:00-04:00",
            "2019-04-29T10:30:00-04:00",
            "2019-04-29T08:00:00-04:00",
          ],
        },
        Object {
          "advisor": "399959",
          "slots": Array [
            "2019-04-29T16:00:00-04:00",
            "2019-04-29T09:30:00-04:00",
            "2019-04-29T11:45:00-04:00",
            "2019-04-29T14:15:00-04:00",
          ],
        },
        Object {
          "advisor": "417239",
          "slots": Array [
            "2019-04-29T18:15:00-04:00",
            "2019-04-29T17:45:00-04:00",
            "2019-04-29T15:45:00-04:00",
          ],
        },
        Object {
          "advisor": "419054",
          "slots": Array [
            "2019-04-29T05:00:00-04:00",
            "2019-04-29T06:00:00-04:00",
            "2019-04-29T03:00:00-04:00",
            "2019-04-29T16:30:00-04:00",
            "2019-04-29T04:00:00-04:00",
            "2019-04-29T07:00:00-04:00",
          ],
        },
        Object {
          "advisor": "424108",
          "slots": Array [
            "2019-04-29T18:00:00-04:00",
            "2019-04-29T16:15:00-04:00",
            "2019-04-29T15:00:00-04:00",
            "2019-04-29T11:15:00-04:00",
            "2019-04-29T12:00:00-04:00",
            "2019-04-29T10:45:00-04:00",
            "2019-04-29T17:00:00-04:00",
          ],
        },
        Object {
          "advisor": "424185",
          "slots": Array [
            "2019-04-29T02:00:00-04:00",
            "2019-04-29T12:45:00-04:00",
            "2019-04-29T10:15:00-04:00",
            "2019-04-29T16:45:00-04:00",
            "2019-04-29T22:00:00-04:00",
            "2019-04-29T23:00:00-04:00",
            "2019-04-29T21:00:00-04:00",
            "2019-04-29T20:00:00-04:00",
            "2019-04-29T13:15:00-04:00",
            "2019-04-29T00:00:00-04:00",
          ],
        },
      ],
    },
    Object {
      "date": "2019-04-27",
      "openings": Array [
        Object {
          "advisor": "424185",
          "slots": Array [
            "2019-04-27T21:45:00-04:00",
            "2019-04-27T23:00:00-04:00",
            "2019-04-27T22:00:00-04:00",
          ],
        },
      ],
    },
    Object {
      "date": "2019-05-01",
      "openings": Array [
        Object {
          "advisor": "319369",
          "slots": Array [
            "2019-05-01T19:00:00-04:00",
          ],
        },
        Object {
          "advisor": "372955",
          "slots": Array [
            "2019-05-01T17:30:00-04:00",
          ],
        },
        Object {
          "advisor": "399958",
          "slots": Array [
            "2019-05-01T19:15:00-04:00",
            "2019-05-01T18:30:00-04:00",
          ],
        },
        Object {
          "advisor": "399959",
          "slots": Array [
            "2019-05-01T14:15:00-04:00",
            "2019-05-01T12:00:00-04:00",
            "2019-05-01T09:00:00-04:00",
            "2019-05-01T13:00:00-04:00",
          ],
        },
        Object {
          "advisor": "419054",
          "slots": Array [
            "2019-05-01T16:30:00-04:00",
            "2019-05-01T05:00:00-04:00",
            "2019-05-01T04:00:00-04:00",
            "2019-05-01T03:00:00-04:00",
            "2019-05-01T07:00:00-04:00",
            "2019-05-01T15:15:00-04:00",
            "2019-05-01T06:00:00-04:00",
          ],
        },
        Object {
          "advisor": "424108",
          "slots": Array [
            "2019-05-01T10:00:00-04:00",
            "2019-05-01T18:00:00-04:00",
            "2019-05-01T14:00:00-04:00",
            "2019-05-01T15:00:00-04:00",
            "2019-05-01T17:00:00-04:00",
            "2019-05-01T11:00:00-04:00",
            "2019-05-01T13:15:00-04:00",
            "2019-05-01T16:00:00-04:00",
          ],
        },
      ],
    },
    Object {
      "date": "2019-05-02",
      "openings": Array [
        Object {
          "advisor": "399958",
          "slots": Array [
            "2019-05-02T17:00:00-04:00",
            "2019-05-02T13:00:00-04:00",
            "2019-05-02T14:00:00-04:00",
            "2019-05-02T16:00:00-04:00",
            "2019-05-02T19:15:00-04:00",
            "2019-05-02T15:30:00-04:00",
            "2019-05-02T12:00:00-04:00",
          ],
        },
      ],
    },
  ],
}
`;
